odoo/addons/base/models/res_partner.py

```diff
@@ -168,6 +168,7 @@ class Partner(models.Model):
     _order = "complete_name ASC, id DESC"
     _rec_names_search = ['complete_name', 'email', 'ref', 'vat', 'company_registry']  # TODO vat must be sanitized the same way for storing/searching
     _allow_sudo_commands = False
+    _check_company_domain = models.check_company_domain_parent_of
```

Comment:
Some domains are using [`=?`](https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html?highlight=search%20domains#search-domains) (`['|', ('company_id', '=?', company_id), ('company_id', '=', False)]`)
So the generic `check_company_domain_parent_of` doesn't work in all cases.

That might also be the reason why some tests fail?

I think that we can always use `=?` because the model should define if `company_id` is required or not, and if it is not defined any value partner should (almost always?) do. If not, I'd add an additional manual constraint.

Reply:
You are right, it looks like an issue. 

Actually, if I remember well, the tests that failed were only with SO.

I don't understand exactly what you mean by "always use `=?`". The `=?` is only used in project app so do I revert the changes in `project.project` and `project.task` ? Or do I add the domain and `check_company` ?

I made some checks, and I was able to access partners of parent company in project and tasks without making any change (I don't really know why, I made the change because it seemed like it was not the case before). So I can just revert the changes in `project.project` and `project.task` ?

Reply:
Update: Actually even with the change I proposed the partners are not accessible by the branch if the company of the project is the branch and it is still impossible to make a project with a partner from a different company, because of `_inverse_company_id()`. 
I will continue investigating but is it necessary to change the project app ?